src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -1085,9 +1222,299 @@ .maxstack 8
+                Diagnostic(ErrorCode.ERR_AssgReadonlyProp, "P").WithArguments("C.P").WithLocation(6, 9),
+                // (9,16): error CS0548: 'C.P': property or indexer must have at least one accessor
+                //     public int P { }
+                Diagnostic(ErrorCode.ERR_PropertyWithNoAccessors, "P").WithArguments("C.P").WithLocation(9, 16),
+                // (13,34): error CS0154: The property or indexer 'C.P' cannot be used in this context because it lacks the get accessor
+                //         System.Console.WriteLine(new C().P);
+                Diagnostic(ErrorCode.ERR_PropertyLacksGet, "new C().P").WithArguments("C.P").WithLocation(13, 34)
+            );
             Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
         }
 
+        [Theory]
+        [InlineData("get;", 0, false)]
+        [InlineData("get; set;", 0, true)]
+        // [InlineData("set;")] PROTOTYPE(semi-auto-props): Not yet supported.
+        [InlineData("get => field;", 1, false)]
+        // [InlineData("get => field; set;")] PROTOTYPE(semi-auto-props): Not yet supported
+        public void TestAssigningFromConstructorThroughBackingField(string accessors, int bindingCount, bool callsSynthesizedSetter)
+        {
+            var comp = CreateCompilation($@"
+public class C
+{{
+    public C()
+    {{
+        P = 5;
+    }}
+
+    public int P {{ {accessors} }}
+
+    public static void Main()
+    {{
+        System.Console.WriteLine(new C().P);
+    }}
+}}
+", options: TestOptions.DebugExe);
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            var expectedCtorIL = callsSynthesizedSetter switch
+            {
+                true => @"
+{
+  // Code size       17 (0x11)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""object..ctor()""
+  IL_0006:  nop
+  IL_0007:  nop
+  IL_0008:  ldarg.0
+  IL_0009:  ldc.i4.5
+  IL_000a:  call       ""void C.P.set""
+  IL_000f:  nop
+  IL_0010:  ret
+}
+",
+                false => @"
+{
+  // Code size       16 (0x10)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""object..ctor()""
+  IL_0006:  nop
+  IL_0007:  nop
+  IL_0008:  ldarg.0
+  IL_0009:  ldc.i4.5
+  IL_000a:  stfld      ""int C.<P>k__BackingField""
+  IL_000f:  ret
+}
+",
+            };
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "5").VerifyIL("C.P.get", @"
+{
+  // Code size        7 (0x7)
+  .maxstack  1
+  IL_0000:  ldarg.0
+  IL_0001:  ldfld      ""int C.<P>k__BackingField""
+  IL_0006:  ret
+}
+").VerifyIL("C..ctor", expectedCtorIL);
+            Assert.Equal(bindingCount, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void TestAssigningFromConstructorThroughSetterWithFieldKeyword_NoGetter()
+        {
+            var comp = CreateCompilation(@"
+public class C
```

Comment:
>class [](http://example.com/codeflow?start=7&length=5)

It feels interesting to test the same scenario for a struct

Reply:
#Pending

Reply:
Updated the test into a `Theory`.